core/model/src/main/java/org/eclipse/rdf4j/model/datatypes/XmlDatatype.java

```diff
@@ -0,0 +1,2470 @@
+	 */
+	public boolean isDecimalDatatype() {
+		return decimal;
+	}
+
+	/**
+	 * Checks whether the supplied datatype is equal to xsd:integer or one of the built-in datatypes that is derived
+	 * from xsd:integer.
+	 *
+	 * @return true if it is an integer type
+	 */
+	public boolean isIntegerDatatype() {
+		return integer;
+	}
+
+	/**
+	 * Checks whether the supplied datatype is equal to xsd:float or xsd:double.
+	 *
+	 * @return true if it is a floating point type
+	 */
+	public boolean isFloatingPointDatatype() {
+		return floatingPoint;
+	}
+
+	/**
+	 * Checks whether the supplied datatype is equal to xsd:dateTime, xsd:date, xsd:time, xsd:gYearMonth, xsd:gMonthDay,
+	 * xsd:gYear, xsd:gMonth or xsd:gDay.These are the primitive datatypes that represent dates and/or times.
+	 *
+	 * @see XMLGregorianCalendar
+	 * @return true if it is a calendar type
+	 */
+	public boolean isCalendarDatatype() {
+		return calendar;
+	}
+
+	/**
+	 * Checks whether the supplied datatype is equal to xsd:duration, xsd:dayTimeDuration, xsd:yearMonthDuration. These
+	 * are the datatypes that represents durations.
+	 *
+	 * @see Duration
+	 * @return true if it is a duration type
+	 */
+	public boolean isDurationDatatype() {
+		return duration;
+	}
+
+	/**
+	 * Checks whether the supplied datatype is ordered.The values of an ordered datatype can be compared to each other
+	 * using operators like <tt>&lt;</tt> and <tt>&gt;</tt>.
+	 *
+	 * @return true if the datatype is ordered
+	 */
+	public boolean isOrderedDatatype() {
+		return isNumericDatatype() || isCalendarDatatype();
+	}
+
+	static HashMap<IRI, XmlDatatype> reverseLookup = new HashMap<>();
+
+	static {
+		for (XmlDatatype value : XmlDatatype.values()) {
+			reverseLookup.put(value.iri, value);
+		}
+	}
+
+	public static XmlDatatype from(IRI datatype) {
+		return reverseLookup.get(datatype);
+	}
+
+	public IRI getIri() {
+		return iri;
+	}
+
+	public int valueComparatorTo(XmlDatatype compareTo) {
+		if (valueComparatorLookup == null) {
+			createLookupTable();
+		}
+
+		return valueComparatorLookup.get(compareTo);
+	}
+
+	private void createLookupTable() {
+		HashMap<XmlDatatype, Integer> valueComparatorLookup = new HashMap<>();
+
+		if (this == DURATION) {
+			valueComparatorLookup.put(DURATION, 0);
+			valueComparatorLookup.put(DATETIME, 20);
+			valueComparatorLookup.put(DATETIMESTAMP, 20);
+			valueComparatorLookup.put(DAYTIMEDURATION, 20);
+			valueComparatorLookup.put(TIME, -16);
+			valueComparatorLookup.put(DATE, 20);
+			valueComparatorLookup.put(GYEARMONTH, -3);
+			valueComparatorLookup.put(GYEAR, -3);
+			valueComparatorLookup.put(GMONTHDAY, -3);
+			valueComparatorLookup.put(GDAY, -3);
+			valueComparatorLookup.put(GMONTH, -3);
+			valueComparatorLookup.put(STRING, -15);
+			valueComparatorLookup.put(BOOLEAN, 2);
+			valueComparatorLookup.put(BASE64BINARY, 2);
+			valueComparatorLookup.put(HEXBINARY, -4);
+			valueComparatorLookup.put(FLOAT, -2);
+			valueComparatorLookup.put(DECIMAL, 16);
```

Comment:
Pre-calculation of `new ValueComparator().compare(d1, d2)` for all datatype combinations.